src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -70,6 +68,7 @@ namespace OpenMS
     public RangeManager<1>,
     public SpectrumSettings
   {
+     friend class MSSpectrumHelper;
```

Comment:
mh, the corresponding code does not look too awkward, so I'm gonna roll with you here. But I'm not sure how soon getting rid of rich peaks will happen. Functionality in this PR is really needed right now for moving forward on the QC front, so I am not eager to wait.